### DISCUSSION TITLE
How to predict link's weight on a directed graph?

### DISCUSSION THREAD

**Discussion by sunshineYin** [pyg-team/pytorch_geometric#2533] [Q&A]
Sorry, I have encountered similar problem.

I am working on a link prediction problem recently. The problem is based on directed graph, and the edges have weights. Here, I want to get nodes' embeddings through gcn like models. And then, to predict the unknown edge's weight between two nodes, based on their embeddings.

So the problem is that the gcns now are mainly for undirected graphs, because of the symmetrical laplace matrix. But, in my problem, the adjancy matrix is not symetrical. So how to fix it, and to predict the edge's weight?

Thanks a lot!

**Comment by rusty1s** [comment]
Please see my reply here: https://github.com/rusty1s/pytorch_geometric/discussions/2391#discussioncomment-697976
